squid-blockexplorer/src/blocks/utils.ts

```diff
@@ -91,38 +91,42 @@ export function createCall(
   });
 }
 
+const PIECE_SIZE = BigInt(1048576);
+const SLOT_PROBABILITY = [BigInt(1), BigInt(6)];
+
 /**
  * Calculates the space pledged by a solution
  * @param {bigint} solutionRange - range of the solution
  * @return {bigint} - space pledged in bytes
  */
 export function calcSpacePledged(solutionRange: bigint): bigint {
-  const MAX_U64 = 2n ** 64n - 1n;
-  const SLOT_PROBABILITY = [1n, 6n];
-  const PIECE_SIZE = 31744n;
-
-  return BigInt(
-    ((MAX_U64 * SLOT_PROBABILITY[0]) / SLOT_PROBABILITY[1] / solutionRange) *
-    PIECE_SIZE
-  );
+  const MAX_U64 = BigInt(2 ** 64 - 1);
+
+  const RECORD_BUCKETS = BigInt(65536);
+  const RECORD_CHUNKS = BigInt(32768);
+  const SOLUTION_RANGE = BigInt(8);
+  const SCALAR = BigInt(32);
+
+  const history_size =
+    (MAX_U64 * SLOT_PROBABILITY[0]) /
+    SLOT_PROBABILITY[1] /
+    (solutionRange * PIECE_SIZE) /
+    (RECORD_BUCKETS * RECORD_CHUNKS) /
+    (SOLUTION_RANGE * SCALAR);
```

Comment:
Why using `()`? It is not grouped in a way that would help understanding it and in fact it destroys the formula.

For example you were supposed to do `/ solutionRange * PIECE_SIZE` and you're doing essentially `/ solutionRange / PIECE_SIZE`, which is will result in a completely different value.

Please be very careful!

As mentioned before, take the actual code (not the high-level formula in the comment) and use that literally (of course you'll replace saturating multiplication with a regular one here).